docs/program-setup/bare_metal/mbed2_porting.md

```diff
@@ -0,0 +1,125 @@
+
+<span class="notes">**Note:** The IAR toolchain does not have a small C library.</span>
+
+- Set the `c_lib` property to indicate which C library should be used by the build tools. If your target has `"default_lib": "small"` defined, then replace it with `"c_lib" : "small"`. Otherwise add `"c_lib" : "small"`. We recommend that ultra constrained devices running bare metal applications link with the small C libraries by default.
+
+<span class="notes">**Note:** If a toolchain does not support a small C library and `c_lib` is set to `small`, the build tools will revert to linking with the standard C library, provided that it is listed in `supported_c_libs` for the given toolchain.</span>
+
+- If `default_toolchain` is set to `uARM`, then replace it with `ARM` and remove `uARM` from `supported_toolchains`. Support for the uARM toolchain, which is the ARM toolchain with microlib, has been removed. Setting `c_lib` to `small` ensures that microlib is used when the ARM toolchain is selected.
+
+<span class="notes">**Note:** As mentioned above, to successfully build with microlib, the target must define a two-region memory model. You might need to replace the content of the TOOLCHAIN_ARM_STD linker file with that of TOOLCHAIN_ARM_MICRO which includes a two-region memory model linker file.</span>
+
+- If your board does not have a low power ticker, ensure that tickless is enabled using the microsecond ticker as follows:
+```
+"overrides": {
+    "tickless-from-us-ticker": true
+}
+```
+
+- It might be necessary to reduce the stack size allocated for your target if it does not have enough RAM. The stack size is configured by setting a value for the `boot-stack-size` attribute; this value must be a multiple of 8 for alignment purposes. By default all targets are configured to have a boot stack size of 0x1000 (4096 bytes) in bare metal. However, this must be overridden if inadequate for your target. We recommend to reduce the boot stack size to 0x400 (1024 bytes) if your target has 8KB of RAM and to 0x300 (768 bytes) if your target has 4KB of RAM.
+```
+"overrides": {
+    "boot-stack-size": "0x400"
+}
+```
+
+### Build and run mbed-os-example-blinky-baremetal
+
+Build the application and program your target: `mbed compile -m <YOUR_TARGET> -t <TOOLCHAIN> --flash --sterm`.
+
+The application running on the target should print a text on the console. Repeat for all supported toolchains.
+
+Troubleshoot any issue.
+
+## Validating the port
+
+To validate the bare metal target configuration, you will execute the Mbed OS greentea test suite with the bare metal configuration. A subset of the tests are automatically skipped by the framework either because the underlying functionality has not been ported to bare metal or because some tests require RTOS features, for examples more complex tests based on multi-threading.
```

Comment:
```suggestion
To validate the bare metal target port, execute the Mbed OS Greentea test suite with the bare metal configuration. This configuration causes Greentea to skip a subset of the tests, either because the underlying functionality has not been ported to bare metal or because some tests require RTOS features (for examples, more complex tests based on multi-threading).
```

Reply:
I changed "configuration" to "port" because there were two configurations in the paragraph and it can be confusing on first reading. You used "port" at the beginning of the document so I assumed it's okay.

Reply:
I will put back configuration but then use bare metal "profile".